src/Controllers/UserController.js

```diff
@@ -60,11 +60,18 @@ export class UserController extends AdaptableController {
       updateFields._email_verify_token_expires_at = {__op: 'Delete'};
     }
 
-    return this.config.database.update('_User', query, updateFields).then((document) => {
-      if (!document) {
-        throw undefined;
-      }
-      return Promise.resolve(document);
+    var self = this;
+    var checkIfAlreadyVerified = new RestQuery(self.config, Auth.master(self.config), '_User', {username: username, emailVerified: true});
```

Comment:
So, I think this is redundant but that the part of that could be removed is the `aUser.emailVerified` in resendVerificationEmail. 

This is the first time in the handling of the `/verify_email` endpoint that the _User object is actually in scope. The PublicAPIRouter hands off the username and token url params to `UserController.verifyEmail` and it's on the controller to get the _User and decide what to do (so the caller can't really do the check, and it would be weird to have the Router interact with the database to do so). In order to decide whether to redirect to the `invalid_verification_link` page or the `verify_email_success` page we need to check if the _User is already verified. 

Note that the `resend_verification_email` endpoint is only going to be hit if we direct to the `invalid_verification_link` page and use it's form to POST to it, so the check for `aUser.emailVerified` in the resendVerificationEmail method seems to be the redundant one, since in the current setup one must pass through `verify_email` to get to a page that allows one to post to `resend_verification_email` (though I think there's an argument for having `resend_verification_email`  encapsulate the check regardless in case it gets called from another place in the future)